crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,144 @@
+    Config,
+    OnlineClient,
+};
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "remove", about = "Remove a contract's code")]
+pub struct RemoveCommand {
+    /// The hash of the smart contract code already uploaded to the chain.
+    #[clap(long, value_parser = parse_code_hash)]
+    code_hash: Option<<DefaultConfig as Config>::Hash>,
+    #[clap(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl RemoveCommand {
+    pub fn is_json(&self) -> bool {
+        self.output_json
+    }
+
+    pub fn run(&self) -> Result<(), ErrorVariant> {
+        let artifacts = self.extrinsic_opts.contract_artifacts()?;
+        let transcoder = artifacts.contract_transcoder()?;
+        let signer = super::pair_signer(self.extrinsic_opts.signer()?);
+
+        let artifacts_path = artifacts.artifact_path().to_path_buf();
+
+        let final_code_hash = match (self.code_hash.as_ref(), artifacts.code.as_ref()) {
+            (Some(code_h), _) => {
+                Ok(code_h.0)
+            }
+            (None, Some(_)) => {
+                artifacts.code_hash()
+            }
+            (None, None) => {
+                Err(
+                    anyhow::anyhow!(
+                    "No code_hash was provided or contract code was not found from artifact file {}. Please provide a code hash with --code-hash argument or specify the path for artifacts files with --manifest-path", 
+                    artifacts_path.display()
+                ))
+            }
+        }?;
+
+        async_std::task::block_on(async {
+            let url = self.extrinsic_opts.url_to_string();
+            let client = OnlineClient::from_url(url.clone()).await?;
+            if let Some(code_removed) = self
+                .remove_code(
+                    &client,
+                    sp_core::H256(final_code_hash),
+                    &signer,
+                    &transcoder,
+                )
+                .await?
+            {
+                let remove_result = code_removed.code_hash;
+
+                if self.output_json {
+                    println!("{}", &remove_result);
+                } else {
+                    name_value_println!("Code hash", format!("{remove_result:?}"));
+                }
+                Result::<(), ErrorVariant>::Ok(())
+            } else {
+                let error_code_hash = hex::encode(final_code_hash);
+                Err(anyhow::anyhow!(
+                    "Error removing the code for the supplied code hash: {}",
+                    error_code_hash
+                )
+                .into())
+            }
+        })
+    }
+
+    async fn remove_code(
+        &self,
+        client: &Client,
+        code_hash: CodeHash,
+        signer: &PairSigner,
+        transcoder: &ContractMessageTranscoder,
+    ) -> Result<Option<api::contracts::events::CodeRemoved>, anyhow::Error> {
```

Comment:
```suggestion
    ) -> Result<Option<api::contracts::events::CodeRemoved>, ErrorVariant> {
```
This will cause a much nicer error message to be displayed, similar to the other commands. 

Before: 

![image](https://user-images.githubusercontent.com/75586/218817901-0f0d1203-9160-442d-a914-43006a8f3897.png)

After: 

![image](https://user-images.githubusercontent.com/75586/218818113-41d094af-4c11-479d-ba47-2e7d95389037.png)

Reply:
Thanks !